services/github/github-version.service.js

```diff
@@ -0,0 +1,239 @@
+'use strict'
+
+const Joi = require('@hapi/joi')
+const { NotFound } = require('..')
+const { addv } = require('../text-formatters')
+const { version: versionColor } = require('../color-formatters')
+const { latest } = require('../version')
+const { GithubAuthService } = require('./github-auth-service')
+const {
+  fetchLatestRelease,
+  releaseInfoSchema,
+} = require('./github-common-fetch')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const queryParamSchema = Joi.object({
+  include_prereleases: Joi.equal(''),
+  include_tags: Joi.equal(''),
+  sort: Joi.string()
+    .valid('date', 'semver')
+    .default('date'),
+}).required()
+
+const tagSchema = Joi.alternatives()
+  .try(
+    Joi.array()
+      .items(
+        Joi.object({
+          name: Joi.string().required(),
+        })
+      )
+      .required(),
+    Joi.array().length(0)
+  )
+  .required()
+
+const releaseInfoArraySchema = Joi.array()
+  .items(releaseInfoSchema)
+  .required()
+
+module.exports = class GithubVersion extends GithubAuthService {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'github/v',
```

Comment:
In light of [GitHub's addition of a Package Registry](https://github.com/features/package-registry), and requests like #3608 to add badge support for GH Package Registry, do you think it would it be helpful to include an additional qualifier in the route to differentiate this badge (versions for releases/tags) from the likely future badge for the package registry (version for a published package on GH)?

I suppose one option would be to leave this as is, with an implicit reference to tags/releases, and then do something like `github/package/v` for the package version. It's admittedly a little trickier to be more explicit here since the whole objective is to remove confusion around tags vs. releases.

Reply:
Great question. Looking at an example of a project that has tags, releases and packages: https://github.com/Codertocat/hello-world-npm/packages

I wonder if actually it would be better to keep tags and releases express it as:

* `/v/release/`
* `/v/tag/`
* `/v/package/`

instead of making `include_tags` a GET param (but keep `include_prereleases` and `sort=date`/`sort=semver` as params).

Reply:
That sounds like a good plan to me!